src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6355,6 +6408,416 @@ register_handler(int sig, struct sigaction *old_handler)
 	}
 }
 
+/* free all EQs allocated for aio */
+
+static void
+free_aio_ctx(void)
+{
+	int i;
+
+	for (i = 0; i < aio_eq_count; i++) {
+		daos_eq_destroy(aio_eq_list[i].eq, 0);
+		D_MUTEX_DESTROY(&aio_eq_list[i].lock);
+	}
+	aio_eq_count = 0;
+}
+
+/* aio functions return negative errno in case of failure */
+static int
+create_ev_eq_for_aio(aio_ctx_t *aio_ctx);
+int
+io_setup(int maxevents, io_context_t *ctxp)
+{
+	int        rc;
+	aio_ctx_t *aio_ctx_obj;
+
+	if (next_io_setup == NULL) {
+		next_io_setup = dlsym(RTLD_NEXT, "io_setup");
+		D_ASSERT(next_io_setup != NULL);
```

Comment:
Is it possible to use pil4dfs with an appliction that does not link libaio, and in which case would this fail?  Perhaps if the user tried that then libaio would be linked anyway as pil4dfs depends on it?

The use of assert here concerns me but it maybe a non-issue.

Reply:
This code is inside io_setup(). If user's application does not use libaio, this function should not be triggered.